tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapCasTest.java

```diff
@@ -104,44 +80,40 @@ public void verify() throws Exception {
         assertEquals("There should not be any data races", 0, failures);
     }
 
-    @Performance
-    public long getOperationCount() {
-        return operations.get();
+    @RunWithWorker
+    public AbstractWorkerTask createBaseWorker() {
+        return new WorkerTask();
     }
 
-    private class Worker implements Runnable {
-        private final Random random = new Random();
+    private class WorkerTask extends SingleOperationAbstractWorkerTask {
         private final Map<Integer, Long> result = new HashMap<Integer, Long>();
 
-        @Override
-        public void run() {
-            for (int k = 0; k < keyCount; k++) {
-                result.put(k, 0L);
+        protected void beforeRun() {
+            if (map.size() != keyCount) {
+                throw new RuntimeException("Warmup has not run since the map is not filled correctly, found size: " + map.size());
             }
 
-            long iteration = 0;
-            while (!testContext.isStopped()) {
-                Integer key = random.nextInt(keyCount);
-                long increment = random.nextInt(100);
-
-                for (; ; ) {
-                    Long current = map.get(key);
-                    Long update = current + increment;
-                    if (map.replace(key, current, update)) {
-                        increment(key, increment);
-                        break;
-                    }
-                }
+            for (int i = 0; i < keyCount; i++) {
+                result.put(i, 0L);
+            }
+        }
 
-                iteration++;
-                if (iteration % logFrequency == 0) {
-                    log.info(Thread.currentThread().getName() + " At iteration: " + iteration);
-                }
-                if (iteration % performanceUpdateFrequency == 0) {
-                    operations.addAndGet(performanceUpdateFrequency);
+        @Override
+        protected void doRun() {
+            Integer key = nextInt(keyCount);
+            long incrementValue = nextInt(100);
+
+            while (true) {
```

Comment:
Can you replace it with for(;;;) Since we use that in the system for cas retries.